indico/modules/events/editing/client/js/components/FileManager/FileList.jsx

```diff
@@ -0,0 +1,152 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useContext, useRef, useState} from 'react';
+import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
+import {Icon} from 'semantic-ui-react';
+import {FileManagerContext, filePropTypes, uploadFiles, deleteFile} from './util';
+import * as actions from './actions';
+
+import './FileManager.module.scss';
+
+function FileAction({onClick, active, icon, className}) {
+  return (
+    <Icon
+      className={className}
+      name={active ? 'spinner' : icon}
+      loading={active}
+      onClick={() => {
+        if (!active) {
+          onClick();
+        }
+      }}
+    />
+  );
+}
+
+FileAction.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  active: PropTypes.bool.isRequired,
+  icon: PropTypes.string.isRequired,
+  className: PropTypes.string.isRequired,
+};
+
+function FileEntry({
+  eventId,
+  fileTypeId,
+  allowMultipleFiles,
+  file: {uuid, filename, state, claimed},
+}) {
+  const ref = useRef(null);
+  const dispatch = useContext(FileManagerContext);
+  const [activeButton, setActiveButton] = useState(null);
+
+  return (
+    <>
+      <span styleName="file-state" className={state || ''}>
+        {filename}
+      </span>
+      <span>
+        {!state && allowMultipleFiles && (
+          <>
+            <FileAction
+              icon="exchange"
+              active={activeButton === 'replace'}
+              styleName="exchange-icon"
+              onClick={() => {
+                ref.current.open();
+              }}
+            />
+            <Dropzone
```

Comment:
What about using the hook version?

Reply:
In this case we needed to use the reference to know which dropzone to open whenever the user clicks the exchange button and it was too complicated with the hook version.

Reply:
We cannot, because we need a `ref` to the `Dropzone`, so that we can trigger the opening of the fiile selection dialog.

Reply:
(what @panagiotappl said :wink:)

Reply:
The callable `open()` like in https://github.com/react-dropzone/react-dropzone/tree/master/examples/file-dialog doesn't work?

Reply:
Oh, the docs don't mention that. Cool.

Reply:
Done.